internal/core/textlayout.rs

```diff
@@ -262,7 +262,7 @@ impl<Length: Clone + Copy + Default + core::ops::AddAssign> TextLine<Length> {
             self.byte_range.start = candidate.byte_range.start;
             self.byte_range.end = self.byte_range.start;
             self.glyph_range.start = candidate.glyph_range.start;
-            self.glyph_range.end = self.glyph_range.end;
+            self.glyph_range.end = candidate.glyph_range.end;
```

Comment:
This is the same as self.glyph_range = candidate.glyph_range;. But this looks suspicious. Was *self=candidate meant?

Reply:
> This is the same as self.glyph_range = candidate.glyph_range;

Yeah, I think that would also be fine (needs clone() to compile though, as Range isn't copy).

> Was *self=candidate meant?

I think the idea here was just to copy the ranges of the graphemes without the whitespace, while a full clone would copy also the trailing whitespace - that would break the match right below.

Either way, I've been spare-time hacking on using this in the GL backend, so that any typographic features we implement don't have to be implemented twice. During that I noticed that this code has bugs in some cases and I've rewritten it with a much simpler version.